### PR TITLE
Get rid of the Impostor.Api dependency

### PR DESCRIPTION
**Pull request by js6pak**

### CODE REVIEW

File: Impostor.Hazel.Abstractions/MessageType.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+﻿namespace Impostor.Hazel.Abstractions;
+
+/// <summary>
+///     Specifies how a message should be sent between connections.
+/// </summary>
+[Flags]
```
Comment (miniduikboot) [inline Impostor.Hazel.Abstractions/MessageType.cs:6]: Why is this [Flags] btw? It was already like this in the old version, but having Unreliable and Reliable at the same time doesn't make sense

File: Hazel/MessageReader.cs [hunk 2]
```diff
@@ -256,21 +249,6 @@ public uint ReadPackedUInt32()
             return output;
         }
 
-        public T ReadNetObject<T>(IGame game) where T : IInnerNetObject
-        {
-            return game.FindObjectByNetId<T>(ReadPackedUInt32());
-        }
-
-        public Vector2 ReadVector2()
```
Comment (miniduikboot) [inline Hazel/MessageReader.cs:264]: moved to an impostor-specific extension method?

**Review by miniduikboot — APPROVED** [review]
just nits

File: Hazel/MessageReader.cs (hunk 2 above)
Comment (js6pak) [reply Hazel/MessageReader.cs:264]: [yup](https://github.com/Impostor/Impostor/blob/net7-nugethazel/src/Impostor.Api/Net/MessageReaderExtensions.cs)

File: Impostor.Hazel.Abstractions/MessageType.cs (hunk 1 above)
Comment (js6pak) [reply Impostor.Hazel.Abstractions/MessageType.cs:6]: ~~Good question, [not for me though](https://github.com/DarkRiftNetworking/Hazel-Networking/blob/master/Hazel/SendOption.cs#L11). I guess we could remove it?~~
It was for long removed upstream fragmentation support, having flags doesn't really change anything here though.

**Pull request merged**